
//testing sync
// console.log('1');
// setTimeout(()=> {
// console.log('2')
// }, 3000);
// console.log('3');


//testing async
// const one = () =>{
//     console.log("Hello");
// }
// const two = () =>{
//     for(i=0; i<= 100; i++){
//         console.log(i);
//     }
//    console.log("Im here")
// }
// const three = ()=>{
//      console.log("World");
// }
// one();
// two();
// three();

//testing promise
// const promiseA = new Promise((resolutionFunc, rejectionFunc) => {
//   resolutionFunc(777);
// });
// // At this point, "promiseA" is already settled.
// promiseA.then((val) => console.log("asynchronous logging has val:", val));
// console.log("immediate logging");

// const promiseA = async () => {
//     return await 777;
// }

// promiseA().then((val)=>{
//     console.log(`asynchronous logging has val: ${val}`)
// })

// console.log("immediate logging")

// const returnNum = () => {
//     return 777;
// }

// const promiseA = async () => {
//     return await returnNum();
// }

// promiseA().then((val)=>{
//     console.log(`asynchronous logging has val: ${val}`)
// });;

// console.log("immediate logging")

